Makefile

```diff
@@ -26,32 +26,48 @@ LDFLAGS=-ldflags "-X main.Commit=${COMMIT} -X main.BuildTime=${BUILD_TIME}"
 build: $(BINARY_SERVER) $(BINARY_CLIENT)
 
 $(BINARY_SERVER): $(SOURCES)
-	go build ${LDFLAGS} -o ${BINARY_SERVER}
+	go build -v ${LDFLAGS} -o ${BINARY_SERVER}
 
 $(BINARY_CLIENT): $(SOURCES)
-	cd ${CLIENT_DIR} && go build -o ../../${BINARY_CLIENT}
-
-deps:
-	go get -u github.com/tools/godep
-	go get -u github.com/jteeuwen/go-bindata/...
-	go get -u github.com/elazarl/go-bindata-assetfs/...
-
-	go get -u github.com/goadesign/goa/...
-	go get -u github.com/goadesign/gorma
+	cd ${CLIENT_DIR} && go build -v -o ../../${BINARY_CLIENT}
 
-generate: $(DESIGNS)
-	goagen bootstrap -d ${PACKAGE_NAME}/${DESIGNDIR}
-	goagen js -d ${PACKAGE_NAME}/${DESIGNDIR} -o assets/ --noexample
-	goagen gen -d ${PACKAGE_NAME}/${DESIGNDIR} --pkg-path=github.com/goadesign/gorma
-	go-bindata-assetfs -debug assets/...
-	godep get
+# These are binary tools from our vendored packages
+$(GOAGEN_BIN):
+	cd $(VENDOR_DIR)/github.com/goadesign/goa/goagen && go build -v
+$(GODEP_BIN):
+	cd $(VENDOR_DIR)/github.com/tools/godep && go build -v
+$(GO_BINDATA_BIN):
+	cd $(VENDOR_DIR)/github.com/jteeuwen/go-bindata/go-bindata && go build -v
+$(GO_BINDATA_ASSETFS_BIN):
+	cd $(VENDOR_DIR)/github.com/elazarl/go-bindata-assetfs/go-bindata-assetfs && go build -v
+$(FRESH_BIN):
+	cd $(VENDOR_DIR)/github.com/pilu/fresh && go build -v
 
 .PHONY: clean
 clean:
-	rm -f ${BINARY_SERVER} && rm -f ${BINARY_CLIENT}
+	# Remove client and server binaries
+	rm -fv $(BINARY_SERVER)
+	rm -fv $(BINARY_CLIENT)
+	# Remove executables from vendored dependencies
+	rm -fv $(GOAGEN_BIN)
+	rm -fv $(GODEP_BIN)
+	rm -fv $(GO_BINDATA_BIN)
+	rm -fv $(GO_BINDATA_ASSETFS_BIN)
+	rm -fv $(FRESH_BIN)
+
+.PHONY: deps
+deps:
+	$(GLIDE_BIN) install
+
+.PHONY: generate
+generate: $(DESIGNS) $(GOAGEN_BIN) $(GO_BINDATA_ASSETFS_BIN) $(GODEP_BIN)
+	$(GOAGEN_BIN) bootstrap -d ${PACKAGE_NAME}/${DESIGN_DIR}
+	$(GOAGEN_BIN) js -d ${PACKAGE_NAME}/${DESIGN_DIR} -o assets/ --noexample
+	$(GOAGEN_BIN) gen -d ${PACKAGE_NAME}/${DESIGN_DIR} --pkg-path=github.com/goadesign/gorma
+	$(GO_BINDATA_ASSETFS_BIN) -debug assets/...
+	$(GODEP_BIN) get
```

Comment:
Unrelated to this line, but I think we should just hard code the PACKAGE_NAME as well. GOPATH can be an array, and we're already referring to a lot of other packages in the Makefile. 

PACKAGE_NAME:="github.com/almighty/almighty-core"